frontend/src/core/utils/fluent/types.js

```diff
@@ -1,8 +1,13 @@
 /* @flow */
 
+export type Identifier = {
+    type: string,
+    name: string,
+};
+
 export type Variant = {
     default: boolean,
-    key: string,
+    key: Identifier,
```

Comment:
`Variant.key` is actually `NumberLiteral | Identifier`. This ties in to my comment about the check for `undefined` as you're treating the poor `NumberLiteral` as if it didn't exist 😢